runtime/lua/vim/_editor.lua

```diff
@@ -715,6 +715,124 @@ function vim._on_key(buf, typed_buf)
   end
 end
 
+-- Decodes a UTF-8 character to a Unicode code point.
```

Comment:
I don't think we want do duplicate the UTF-8 implementation in lua. A tight, simple "inner loop" like mb_utf_index_to_bytes is the kind of code which makes sense to keep in C.

Doing the argument parsing and validation in lua makes sense, but then it could shell out to a private, smaller, lua wrapper around `mb_utf_index_to_bytes` .

Reply:
I did it in Lua as thats what I know and kept the whole implementation more tightly coupled.

Happy to do as you suggest and reach to C, but I might need some help with how best to action it

Reply:
As another thought. I could move the string to codepoint lua code to ```vim.text``` and potentially make it public for use it in other places

It is also used in my PR for ```str_utfindex``` #30600

I do think that having the implementation all in pure lua is simpler